app/src/lib/download.ts

```diff
@@ -2,19 +2,33 @@ import { getAuthHeader } from "./auth/getAuthHeader";
 import appConfig from "../config/appConfig";
 import { FileMetadata } from "../types";
 
-export const getZipUrl = (packetId: string, files: FileMetadata[]) => {
-  const paths = files.map((file) => file.path);
+const filesToPathsParam = (files: FileMetadata[]) =>
+  files.map((file) => `paths=${encodeURIComponent(file.path)}`).join("&");
 
-  return `${appConfig.apiUrl()}/packets/${packetId}/zip?paths=${encodeURIComponent(paths.join(","))}`;
-};
+const filesUrl = (packetId: string, files: FileMetadata[], token: string, filename: string, inline = false) =>
+  `${appConfig.apiUrl()}/packets/${packetId}/files?` +
+  `${filesToPathsParam(files)}&token=${token}&filename=${filename}&inline=${inline}`;
 
-export const getFileUrl = (file: FileMetadata, packetId: string, inline = false) =>
-  `${appConfig.apiUrl()}/packets/${packetId}/file?hash=${file.hash}&filename=${file.path}&inline=${inline}`;
+const getOneTimeToken = async (packetId: string, files: FileMetadata[], filename: string) => {
+  const res = await fetch(`${appConfig.apiUrl()}/packets/${packetId}/files/token?${filesToPathsParam(files)}`, {
+    method: "POST",
+    headers: getAuthHeader()
+  });
+  const json = await res.json();
 
-export const getFileObjectUrl = async (url: string, filename: string) => {
-  const headers = getAuthHeader();
-  const res = await fetch(url, { method: "GET", headers });
+  if (!res.ok) {
+    const msg = json.error?.detail ? `Error: ${json.error.detail}` : `Error retrieving token for ${filename}`;
+    throw new Error(msg);
+  }
 
+  return json.id;
+};
+
+// Fetch a file and pack it into a blob that is stored in the browser’s memory at a URL, until revoked.
+export const getFileObjectUrl = async (file: FileMetadata, packetId: string, filename: string, inline = true) => {
+  const token = await getOneTimeToken(packetId, [file], filename);
```

Comment:
We could also make it so that the token is optional in the files endpoint if a JWT bearer token is provided, then we wouldn't need a OTT for this code path.